rest/data/host.go

```diff
@@ -244,18 +244,16 @@ func (dbc *MockConnector) FindHostByIdWithOwner(hostID string, user gimlet.User)
 
 func findHostByIdWithOwner(c Connector, hostID string, user gimlet.User) (*host.Host, error) {
 	host, err := c.FindHostById(hostID)
+	if host == nil {
+		return nil, err
```

Comment:
I bet that err is going to be nil in this case. should we build/return a gimlet.ErrorResponse with a 404 here?